src/flow-manager.c

```diff
@@ -955,7 +1007,30 @@ static TmEcode FlowManager(ThreadVars *th_v, void *thread_data)
         memset(&sleep_startts, 0, sizeof(sleep_startts));
         gettimeofday(&sleep_startts, NULL);
 #endif
-        usleep(250);
+
+        if (emerg || !time_is_live) {
+            usleep(250);
+        } else {
+            struct timeval cond_tv;
+            gettimeofday(&cond_tv, NULL);
+            struct timeval add_tv;
+            add_tv.tv_sec = 0;
+            add_tv.tv_usec = (sleep_per_wu * 1000);
```

Comment:
we shouldn't backport the work unit stuff either, just the cond commits. They should sleep based on `next_run_ms` I suppose.

Reply:
Ok ... Will rework and submit a new PR.